lib/datasets/house-of-representative.rb

```diff
@@ -0,0 +1,128 @@
+                        :house_of_representatives_of_attitude_of_in_house_group_during_deliberation,
+                        :house_of_representatives_of_support_in_house_group_during_deliberation,
+                        :house_of_representatives_of_opposition_in_house_group_during_deliberation,
+                        :house_of_councillors_of_accepted_bill_on_preliminary_consideration,
+                        :house_of_councillors_of_preliminary_refer_on,
+                        :house_of_councillors_of_preliminary_refer_commission,
+                        :house_of_councillors_of_accepted_bill_on,
+                        :house_of_councillors_of_refer_on,
+                        :house_of_councillors_of_refer_commission,
+                        :house_of_councillors_of_finished_consideration_on,
+                        :house_of_councillors_of_consideration_result,
+                        :house_of_councillors_of_finished_deliberation_on,
+                        :house_of_councillors_of_deliberation_result,
+                        :promulgated_on,
+                        :law_number,
+                        :submitters,
+                        :supporters_of_submitted_bill)
+
+    JAPANESE_DATE_MATCHER = /\A(平成|令和)\s*(\d{1,2}|元)年\s*\d{1,2}月\s*\d{1,2}日\z/.freeze
+    SPLIT_COLUMN_CHAR_ON_HEADER = "／".freeze
+    SPLIT_COLUMN_CHAR_ON_FIELD = SPLIT_COLUMN_CHAR_ON_HEADER
+
+    def initialize
+      super()
+
+      @metadata.id = "house-of-representative"
+      @metadata.name = "Bill of the House of Representatives of Japan"
+      @metadata.url = "https://smartnews-smri.github.io/house-of-representatives"
+      @metadata.licenses = ["MIT"]
+      @metadata.description = "Bill of the House of Representatives of Japan"
+    end
+
+    def each
+      return to_enum(__method__) unless block_given?
+
+      open_data do |csv|
+        csv.each do |row|
+          row = split_csv_column(row)
+          row.fields.each_with_index do |field, idx|
+            row[idx] = parse_japanese_date(field)
+          end
+          %w(議案提出会派 衆議院審議時賛成会派 衆議院審議時反対会派 議案提出者一覧 議案提出の賛成者).each do |array_column_name|
+            row[array_column_name] = parse_array(row[array_column_name])
+          end
+          record = Record.new(*row.fields)
+          yield(record)
+        end
+      end
+    end
+
+    private
+
+    def open_data
+      data_url = "https://raw.githubusercontent.com/smartnews-smri/house-of-representatives/main/data/gian.csv"
+      data_path = cache_dir_path + "bill.csv"
+      download(data_path, data_url)
+
+      CSV.open(data_path, col_sep: ",", headers: true, converters: %i(integer)) do |csv|
+        yield(csv)
+      end
+    end
+
+    def parse_array(column_value)
+      column_value.to_s.split("; ")
+    end
+
+    def parse_japanese_date(field)
+      return field unless field.to_s.match?(JAPANESE_DATE_MATCHER)
+      Date.parse(normalize_jisx0301_format(field))
```

Comment:
I think that using `MatchData` is better instead of `matched?` and normalizing string:

```ruby
case field
when nil
  nil
when /\A(平成|令和)\s*(\d{1,2}|元)年\s*(\d{1,2})月\s*(\d{1,2})日\z/
  match_data = Regex.last_match
  # TODO: Make this a constant
  era_initials = {
    "平成" => "H",
    "令和" => "R",
  }
  era_initial = era_initials[match_data[1]]
  year = match_data[2]
  if year == "元"
    year = "01"
  else
    year = year.rjust(2, "0")
  end
  month = match_data[3].rjust(2, "0")
  day = match_data[4].rjust(2, "0") 
  Date.jisx0301("#{era_initial}#{year}.#{month}.#{day}")
else
  field # or raise?
end
```

Reply:
I don't know `MatchData`. Thank you for the clear code.

Reply:
Added at 3e114f7.